File: RepoTransBench/target_projects/Python/C++/scrapinghub_adblockparser/public_tests/test_public_rule_types.cpp
#include <gtest/gtest.h>
#include "adblockparser.h"

TEST(PublicRuleTypesTest, PublicIsComment) {
    std::vector<std::string> COMMENT_RULES = {
        "! This is a comment line",
        "! Title: Example Filter List",
        "! Expires: 4 days",
        "! Homepage: https://example.com/",
        "[Adblock]",
        "!#include another_list.txt"
    };
    for(const auto& text : COMMENT_RULES) {
        AdblockRule rule(text);
        EXPECT_TRUE(rule.is_comment());
        EXPECT_FALSE(rule.is_html_rule());
        EXPECT_FALSE(rule.is_exception());
        EXPECT_TRUE(rule.options().empty());
        EXPECT_FALSE(rule.has_regex());
    }
}

TEST(PublicRuleTypesTest, PublicIsHtmlRule) {
    std::vector<std::string> HTML_RULES = {
        "##.bannerAd",
        "@@##.sponsoredContent",
        "mysite.com#@##sidebar",
        "@@##.cookieBar",
        "example.net,example.org#@##promo",
        "##a[href^='https://tracker.example.com/']",
        "##img[src$='.ads.png']"
    };
    for(const auto& text : HTML_RULES) {
        AdblockRule rule(text);
        EXPECT_TRUE(rule.is_html_rule());
        EXPECT_FALSE(rule.is_comment());
    }
}

File: RepoTransBench/target_projects/Python/Rust/scrapinghub_adblockparser/public_tests/test_public_rule_types.rs
use scrapinghub_adblockparser::adblockparser::AdblockRule;

const COMMENT_RULES: &[&str] = &[
    "! This is a comment line",
    "! Title: Example Filter List",
    "! Expires: 4 days",
    "! Homepage: https://example.com/",
    "[Adblock]",
    "!#include another_list.txt",
];

const HTML_RULES: &[&str] = &[
    "##.bannerAd",
    "@@##.sponsoredContent",
    "mysite.com#@##sidebar",
    "@@##.cookieBar",
    "example.net,example.org#@##promo",
    "##a[href^='https://tracker.example.com/']",
    "##img[src$=\".ads.png\"]",
];

#[test]
fn test_public_is_comment() {
    for &text in COMMENT_RULES {
        let rule = AdblockRule::new(text);
        assert!(rule.is_comment);
        assert!(!rule.is_html_rule);
        assert!(!rule.is_exception);
        assert!(rule.options.is_empty());
        assert!(rule.regex.is_none());
    }
}

#[test]
fn test_public_is_html_rule() {
    for &text in HTML_RULES {
        let rule = AdblockRule::new(text);
        assert!(rule.is_html_rule);
        assert!(!rule.is_comment);
    }
}